include/oneapi/dpl/internal/function.h

```diff
@@ -90,6 +91,123 @@ get_access(const Policy& policy, T* ptr)
     assert(sycl::get_pointer_type(ptr, policy.queue().get_context()) == sycl::usm::alloc::shared);
     return ptr;
 }
+
+template <typename ValueType, typename Policy, typename Iterator>
+ValueType
+get_data_at(const Policy& p, Iterator i, size_t index)
+{
+    return get_access<sycl::access::mode::read>(p, i)[index];
+}
+
+template <typename ValueType, typename Policy, typename T>
+ValueType
+get_data_at(const Policy& policy, T* ptr, size_t index)
+{
+    sycl::queue q = policy.queue();
+
+    switch (sycl::get_pointer_type(ptr, q.get_context()))
+    {
+    case sycl::usm::alloc::host:
+    case sycl::usm::alloc::shared:
+        return ptr[index];
+
+    case sycl::usm::alloc::device:
+    {
+        ValueType host_data;
```

Comment:
This line and line 127 (`return {}`) requires a default constructor for type  `ValueType`. Can we avoid such additional requirement?

Reply:
Fixed. Please check.